### PR TITLE
Re-working registers and peripherals

### PR DESCRIPTION
**Pull request by thejpster**
As discussed in https://github.com/rust-embedded/book/issues/46

### CODE REVIEW

**Comment by thejpster** [comment]
I think I'm done? Feedback welcome!

File: src/start/registers.md [hunk 1]
```diff
@@ -1,143 +1,186 @@
...
-same piece of RAM, one after the other, the compiler can notice this and just
-skip the first write entirely. In C, we can mark variables as `volatile` to
-ensure that every read or write occurs as intended. In Rust, we instead mark
-the *accesses* as volatile, not the variable.
+The `read()` function takes a closure with a single argument. Typically we call this `r`. This argument then gives read-only access to the various sub-fields within this register, as defined by the manufacturer's SVD file for this chip. You can find all the functions available on the 'r' for this particular register, in this particular peripheral, on this particular chip, in the [tm4c123x documentation].
 
 ```rust
-let systick = unsafe { &mut *(0xE000_E010 as *mut SysTick) };
-let time = unsafe { std::ptr::read_volatile(&mut systick.cvr) };
+if pwm.ctl.read(|r| r.globalsync0().is_set()) {
```
[the remark this answers is not in the document]
Comment (thejpster) [reply src/start/registers.md]: D'oh!

File: src/peripherals/a-first-attempt.md [hunk 2]
```diff
@@ -1,97 +1,139 @@
...
 
-fn something_else() {
-    let mut serial = SerialPort::new();
-    // We gotta go fast for this!
-    serial.write_speed(SerialPort::SER_PORT_SPEED_HI);
-    // send some data...
+pub fn example_usage() -> String {
+    let mut st = SystemTimer::new();
+    st.set_reload(0);
+    format!("Time is now 0x{:08x}", st.get_time())
```
[the remark this answers is not in the document]
Comment (thejpster) [reply src/peripherals/a-first-attempt.md]: Not sure the closure example helps with the example that immediately follows. Is this better?

**Comment by jamesmunns** [comment]
@japaric What I wanted to cover in `rusty.md` is more or less covered in `src/start/registers.md` now - I intended to show how to interact with `-pac` and microarchitecture crates, which I think @thejpster has already done and excellent job with.

I'm OK to merge this as-is now.

File: src/peripherals/a-first-attempt.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (japaric) [reply src/peripherals/a-first-attempt.md]: :+1:

**Pull request merged**